addons/mail/static/src/model/model_field_relation_set.js

```diff
@@ -49,93 +49,111 @@ export class RelationSet {
+                            if (valA === valB) {
+                                break;
+                            }
+                            if (!valA) {
+                                return -1;
+                            }
+                            if (!valB) {
+                                return 1;
+                            }
+                            break;
+                        }
+                        case 'case-insensitive-asc': {
+                            if (typeof valA !== 'string' || typeof valB !== 'string') {
+                                break;
+                            }
+                            const cleanedValA = cleanSearchTerm(valA);
+                            const cleanedValB = cleanSearchTerm(valB);
+                            if (cleanedValA === cleanedValB) {
+                                break;
+                            }
+                            return cleanedValA < cleanedValB ? -1 : 1;
+                        }
+                        case 'smaller-first':
+                            if (typeof valA !== 'number' || typeof valB !== 'number') {
+                                break;
+                            }
+                            if (valA === valB) {
+                                break;
+                            }
+                            return valA - valB;
+                        case 'greater-first':
+                            if (typeof valA !== 'number' || typeof valB !== 'number') {
+                                break;
+                            }
+                            if (valA === valB) {
+                                break;
+                            }
+                            return valB - valA;
+                        case 'most-recent-first':
+                            if (!(valA instanceof Date) || !(valB instanceof Date)) {
+                                break;
+                            }
+                            if (valA === valB) {
+                                break;
+                            }
+                            return valB - valA;
+                        default:
+                            throw Error(`sort compare method "${compareMethod}" is not supported.`);
+                    }
+                }
+                return 0;
+            };
+            const search = (from, to) => {
+                if (from === to) {
+                    return to;
+                }
+                const m = Math.floor((from + to) / 2);
+                const compare = compareFunction(this.sortArray[m], value);
+                if (compare > 0) {
+                    return search(from, m);
+                }
+                if (compare < 0) {
+                    return search(m + 1, to);
+                }
+                return m;
+            };
+            // insert correct position
+            this.sortArray.splice(search(0, this.sortArray.length), 0, value);
```

Comment:
It's strange to have `0` and `length` as boundaries, array code typically uses `length - 1` when using 0-index.

Reply:
Ok, makes sense, because sometimes we need to insert at the end (new index) and not at an existing position.